modules/swagger-codegen/src/main/resources/typescript-angular2/api.mustache

```diff
@@ -34,6 +34,21 @@ export class {{classname}} {
             this.configuration = configuration;
         }
     }
+	
+	/**
+     * 
+     * Extends object by coping non-existing properties.
+     * @param objA object to be extended
+     * @param objB source object
+     */
+    private extendObj<T1,T2>(objA: T1, objB: T2) {
```

Comment:
@Vrolijkx @wing328 I did it in this way because it was like that in angular-typescript. Totally agree it's not super cool. I will find time to refactor it and implement consumes / produces mentioned above. 
Thanks!